pipelines/broad/arrays/imputation/Imputation.wdl

```diff
@@ -0,0 +1,339 @@
+version 1.0
+
+import "../../../../structs/imputation/ImputationStructs.wdl" as structs
+import "../../../../tasks/broad/ImputationTasks.wdl" as tasks
+
+workflow ImputationPipeline {
+	input {
+		Int chunkLength = 25000000
+		Int chunkOverlaps = 5000000 # this is the padding that will be added to the beginning and end of each chunk to reduce edge effects
+
+		# You can either input a multisample VCF or an array of single sample VCFs
+		# The pipeline will just merge the single sample VCFs into one multisample VCF
+		# and then impute the multisample VCF
+		# If you want to run a single sample VCF, set the multi_sample_vcf input to the
+		# single sample VCF
+		File? multi_sample_vcf
+		File? multi_sample_vcf_index
+		Array[File]? single_sample_vcfs
+		Array[File]? single_sample_vcf_indices
+
+		Boolean perform_extra_qc_steps # these are optional additional extra QC steps from Amit's group that should only be
+		# run for large sample sets, especially a diverse set of samples (it's further limiting called at sites to 95% and by HWE)
+		File ref_dict # for reheadering / adding contig lengths in the header of the ouptut VCF, and calculating contig lengths
+		Array[ReferencePanelContig] referencePanelContigs
+		String genetic_maps_eagle = "/genetic_map_hg19_withX.txt.gz" # this is for Eagle, it is in the docker image
+		String output_callset_name # the output callset name
+		Boolean split_output_to_single_sample = false
+		File haplotype_database
+		Int merge_ssvcf_mem = 3 # the memory allocation for MergeSingleSampleVcfs (in GiB)
+
+		Float frac_well_imputed_threshold = 0.9 # require fraction of sites well imputed to be greater than this to pass
+		Float chunks_fail_threshold = 1 # require fewer than this many chunks to fail in order to pass
+	}
+	# Docker images here
+	String bcftools_docker_tag = "us.gcr.io/broad-dsde-methods/imputation_bcftools_vcftools_docker:v1.0.0" #"us.gcr.io/broad-dsde-methods/bcftools:v1.9-1-deb_cv1"
+	String bcftools_vcftools_docker_tag = "us.gcr.io/broad-dsde-methods/imputation_bcftools_vcftools_docker:v1.0.0" #"us.gcr.io/broad-dsde-methods/skwalker-imputation:with_vcftools" # TODO: use a public one (not suse bcftools biocontainers also has vcftools )
+	String gatk_docker_tag = "us.gcr.io/broad-gatk/gatk:4.1.9.0"
+	String minimac4_docker_tag = "us.gcr.io/broad-dsde-methods/imputation-minimac-docker:v1.0.0" #"us.gcr.io/broad-dsde-methods/skwalker-imputation:test" # this has the exact version of minimac and eagle we want to perfectly match Michigan Server
+	String eagle_docker_tag = "us.gcr.io/broad-dsde-methods/imputation_eagle_docker:v1.0.0"
+	String ubuntu_docker_tag = "ubuntu:20.04"
+	String rtidyverse_docker_tag = "rocker/tidyverse:4.1.0"
+
+	if (defined(single_sample_vcfs)) {
```

Comment:
Maybe this should be if !defined(multi_sample_vcf) - the select_first's you are using below would take that first, and so if *both* single samples and multi-samples were provided it would do an unnecessary merge.

Reply:
I think both inputs being defined should not be allowed, as it is likely an indication of a mistake, and the user could easily imputing the wrong data because they didn't realize they had passed data via both input methods.  So I'll add a method to check that one and only one input method is being used, and fail the pipeline with a message otherwise.